pkg/storage/backend-storage/backend-storage.go

```diff
@@ -71,24 +74,128 @@ func IsBackendStorageNeededForVM(vm *corev1.VirtualMachine) bool {
 	return HasPersistentTPMDevice(&vm.Spec.Template.Spec)
 }
 
-func CreateIfNeeded(vmi *corev1.VirtualMachineInstance, clusterConfig *virtconfig.ClusterConfig, client kubecli.KubevirtClient) error {
+type BackendStorage struct {
+	client          kubecli.KubevirtClient
+	clusterConfig   *virtconfig.ClusterConfig
+	scStore         cache.Store
+	accessModeCache map[string]v1.PersistentVolumeAccessMode
+	cacheLock       sync.Mutex
+}
+
+func NewBackendStorage(client kubecli.KubevirtClient, clusterConfig *virtconfig.ClusterConfig, scStore cache.Store) *BackendStorage {
+	return &BackendStorage{
+		client:          client,
+		clusterConfig:   clusterConfig,
+		scStore:         scStore,
+		accessModeCache: map[string]v1.PersistentVolumeAccessMode{},
+	}
+}
+
+func (bs *BackendStorage) getStorageClass() (string, error) {
+	storageClass := bs.clusterConfig.GetVMStateStorageClass()
+	if storageClass != "" {
+		return storageClass, nil
+	}
+
+	for _, obj := range bs.scStore.List() {
+		sc := obj.(*storagev1.StorageClass)
+		if sc.Annotations["storageclass.kubernetes.io/is-default-class"] == "true" {
+			return sc.Name, nil
+		}
+	}
+
+	return "", fmt.Errorf("no default storage class found")
+}
+
+func (bs *BackendStorage) getMode(storageClass string, mode v1.PersistentVolumeMode) v1.PersistentVolumeAccessMode {
+	bs.cacheLock.Lock()
+	defer bs.cacheLock.Unlock()
+	accessMode, exists := bs.accessModeCache[storageClass]
```

Comment:
I tend to avoid inlining actions in `if` conditions, I think it hurts readability.
However, if it's considered best practice or if you feel strongly about it, I can make the change.

Reply:
Thank you, will change!

Reply:
mmh, I think you are simply missing a `:`. This compiles from your branch:
```diff
diff --git a/pkg/storage/backend-storage/backend-storage.go b/pkg/storage/backend-storage/backend-storage.go
index 7c3dc58fc1..87650493c7 100644
--- a/pkg/storage/backend-storage/backend-storage.go
+++ b/pkg/storage/backend-storage/backend-storage.go
@@ -110,8 +110,7 @@ func (bs *BackendStorage) getStorageClass() (string, error) {
 func (bs *BackendStorage) getMode(storageClass string, mode v1.PersistentVolumeMode) v1.PersistentVolumeAccessMode {
        bs.cacheLock.Lock()
        defer bs.cacheLock.Unlock()
-       accessMode, exists := bs.accessModeCache[storageClass]
-       if exists {
+       if accessMode, exists := bs.accessModeCache[storageClass]; exists {
                return accessMode
        }
 
@@ -138,7 +137,7 @@ func (bs *BackendStorage) getMode(storageClass string, mode v1.PersistentVolumeM
                if property.VolumeMode == nil || *property.VolumeMode != mode || property.AccessModes == nil {
                        continue
                }
-               for _, accessMode = range property.AccessModes {
+               for _, accessMode := range property.AccessModes {
                        switch accessMode {
                        case v1.ReadWriteMany:
                                return v1.ReadWriteMany
```

Reply:
My bad, it does, I saw the error for the second one and assumed it was about the first. Ok fine then, sorry about that :)